### ISSUE TITLE
Create a Publicly-Readable Challenge Period in the SMC for Notaries

### ISSUE THREAD

**Issue by rauljordan** [prysmaticlabs/geth-sharding#166]
An important part of the Minimal Sharding Protocol and our first release is enforcing a challenge period 
for a Notary's Proof of Custody. That is, when a Notary votes on a collation header and submits a proof of custody to show that it has the full collation data at time of submission, we want to prevent notaries from deleting this data from their persistent storage until a certain challenge period is over.

This is critical for sync as when notaries are selected on a shard in a period, they will have to sync the latest 25 collations for said shard and also fetch the submitted headers to the SMC by proposers. The windback process for fetching collations will be synced from other notaries that have persisted this data due to the challenge period requirement.

We need a publicly-readable challenge period that our sharding node can enforce on Notaries and notify them when it is safe to delete persisted data (when the challenge period is over).

Also implement the ability for a notary service to read this from the SMC on every rerun of its event loop and eventually help delete persisted data automatically when the period is over.

Assigning provisionally to @terenc3t.

**Comment by terenc3t** [comment]
How do we we calculate the challenge period? is there a formula? or it's just constant 25 periods?

**Comment by rauljordan** [comment]
It should be the same as the windback length I believe

**Comment by terenc3t** [comment]
That's 25 periods. What does the SMC needs to do?  If we know challenge period is constant at 25 periods. Can the client calculate what to delete on its own?
```
if currentPeriod - downloadedPeriod > 25 {
  delete collation[downloadedPeriod]
}
```

**Comment by terenc3t** [comment]
Got it. Addressed in #168. Can you take a look when you have a chance?

**Comment by terenc3t** [comment]
Closed via #168
